packages/teleport/src/AppLauncher/useAppLauncher.ts

```diff
@@ -31,9 +32,15 @@ export default function useAppLauncher() {
         // make a redirect to the requested app auth endpoint
         const location = window.location;
         const port = location.port ? ':' + location.port : '';
-        window.location.replace(
-          `https://${result.fqdn}${port}/x-teleport-auth#value=${result.value}`
-        );
+        const state = getUrlParameter('state', location.search);
+        const authUrl = `https://${result.fqdn}${port}/x-teleport-auth`;
+        if (state === '') {
+          const clusterId = params.clusterId ? params.clusterId : '';
+          const publicAddr = params.publicAddr ? params.publicAddr : '';
+          window.location.replace(`${authUrl}?cluster=${clusterId}&addr=${publicAddr}`);
+        } else {
+          window.location.replace(`${authUrl}?state=${state}#value=${result.value}`);
+        }
```

Comment:
```ts
if (state === '') {
   const clusterId = params.clusterId ? params.clusterId : '';
   const publicAddr = params.publicAddr ? params.publicAddr : '';
   window.location.replace(`${authUrl}?cluster=${clusterId}&addr=${publicAddr}`);
 } else {
  window.location.replace(`${authUrl}?state=${state}#value=${result.value}`);
}
```

@kimlisa could you re-write above code with below? 

```ts
let url = ${authUrl}?state=${state}#value=${result.value}`;
if (!state){
  url = `${authUrl}?cluster=${params.clusterId}&addr=${params.publicAddr}`
}

window.location.replace(url);
```

Reply:
```js
const url = new URL(`https://{fqdn});
if (state) {
  url.searchParams.set("cluster", params.clusterId)
  url.searchParams.set("addr", params.publicAddr)
}
url.hash = `#value=${result.value}`;

window.location.replace(s)
```

Reply:
I'd rather not pass valid sessions around unless necessary, i.e. set `#value=` only if `state` is set as it's done in my code.